shared/sync_string_set.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2017 Percona LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shared
+
+import "sync"
+
+// String set synchronized with sync.RWMutex
+type SyncStringSet struct {
+	m  map[string]struct{}
+	mu sync.RWMutex
+}
+
+// NewSyncStringSet returns new SyncStringSet
+func NewSyncStringSet() *SyncStringSet {
+	return &SyncStringSet{
+		m:  make(map[string]struct{}),
+		mu: sync.RWMutex{},
```

Comment:
Please never write `sync.RWMutex{}` or `sync.WaitGroup{}`:
1. Zero values of those types are useful. No initialisation is needed.
2. There is no struct field you can initialise.
3. In fact, it is not even important that it is a struct.

In that case, just drop that line.

Reply:
Done